src/main/java/es/roobre/chestorganizer/ChestOrganizer.java

```diff
@@ -13,60 +23,73 @@
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.plugin.java.JavaPlugin;
 
-import java.util.*;
-import java.util.Comparator;
-
 public final class ChestOrganizer extends JavaPlugin implements Listener {
+	private Logger log = getLogger();
+
+	// TODO: move these configs out to a config file
     private static final int RANGE_HORIZONTAL = 5;
     private static final int RANGE_VERTICAL = 1;
 
-    private static final HashSet<Material> ACTIVATOR_MATERIALS = new HashSet<>(Arrays.asList(Material.REDSTONE_BLOCK));
+    private static final Set<Material> ACTIVATOR_MATERIALS = Stream.of(Material.REDSTONE_BLOCK).collect(Collectors.toSet());    
+    private static final Set<Material> ACTIVATOR_CONTAINERS = Stream.of(Material.CHEST).collect(Collectors.toSet());
 
     @Override
     public void onEnable() {
-        getLogger().info("ChestOrganizer enabled, registering listeners...");
+        log.info("ChestOrganizer enabled, registering listeners...");
+        
         getServer().getPluginManager().registerEvents(this, this);
     }
 
     @EventHandler
     public void onInventoryClick(InventoryClickEvent click) {
         // Get currently open inventory
         InventoryHolder holder = click.getInventory().getHolder();
-        if (!(holder instanceof Chest)) {
+        
+        if (!(isOrganizer(holder))) {
             return;
         }
 
         // Get clicked inventory (which can be a player's, if they shift-clicked an item
         Inventory clickedInventory = click.getClickedInventory();
-        if (clickedInventory == null) return; // Should not happen but just to make the linter happy
+        
+        // Should not happen but just to make the linter happy
+        if (clickedInventory == null) {
+        	return; 
+        }
+        
         InventoryHolder clickedHolder = clickedInventory.getHolder();
 
         ItemStack playerStack;
         InventoryAction action = click.getAction();
 
-        if (clickedHolder instanceof Chest && (action == InventoryAction.PLACE_ALL || action == InventoryAction.PLACE_ONE || action == InventoryAction.PLACE_SOME)) {
+        if (isOrganizer(clickedHolder) && (action == InventoryAction.PLACE_ALL || action == InventoryAction.PLACE_ONE || action == InventoryAction.PLACE_SOME)) {
```

Comment:
Maybe a better alternative would be to check if `clickedHolder` is the same as `holder`, instead of do the longer `isOrganizer` check twice